### ISSUE TITLE
Issue in removing nan's from the trend variable

### ISSUE THREAD

**Issue by ashukl** [statsmodels/statsmodels#4196]
Hi,
I am trying to use your code given in the link: https://github.com/statsmodels/statsmodels/blob/master/statsmodels/tsa/seasonal.py
to remove the nan's which occur due to the moving averages. When i put the variable extrapolate_trend >0 then i get a value error saying: 
ValueError: could not broadcast input array from shape (26,1) into shape (26)
I want to extrapolate the trend values for a time series at a week level

**Comment by josef-pkt** [comment]
Can you make up a full example that I can run, and show the full traceback, at least the end with the exception line.

There might be something wrong if the series is a one-column array, but I cannot see anything related from skimming the code.

**Comment by ashukl** [comment]
If i use the data mentioned in the end of the code itself (which is a one dimensional array):
x = np.array([-50, 175, 149, 214, 247, 237, 225, 329, 729, 809,
                  530, 489, 540, 457, 195, 176, 337, 239, 128, 102,
                  232, 429, 3, 98, 43, -141, -77, -13, 125, 361, -45, 184])
    results = seasonal_decompose(x, freq=4, extrapolate_trend = 4)
I get an error saying ValueError: could not broadcast input array from shape (2,1) into shape (2)

Would i have to convert it to a 2 column array?

**Comment by josef-pkt** [comment]
Thanks for reporting this. It needs to be fixed before the upcoming release.

With two column it works, but this is a bug in `_extrapolate_trend` which doesn't work with 1-D array.
Because the x is squeezed in the function, using a 2-D single column is treated the same way as a 1-D and doesn't provide a workaround.

```
from statsmodels.tsa.seasonal import seasonal_decompose, _extrapolate_trend

_extrapolate_trend(np.arange(12), 4)
Traceback (most recent call last):

  File "<ipython-input-38-235087ac0eb0>", line 1, in <module>
    _extrapolate_trend(np.arange(12), 4)

  File "m:\...\statsmodels\tsa\seasonal.py", line 37, in _extrapolate_trend
    trend[:front] = (np.arange(0, front) * np.c_[k] + np.c_[n]).T

ValueError: could not broadcast input array from shape (0,1) into shape (0)


_extrapolate_trend(np.arange(12)[:, None], 4)
Out[39]: 
array([[ 0],
       [ 1],
       [ 2],
       ..., 
       [ 9],
       [10],
       [11]])
```

**Comment by josef-pkt** [comment]
I'm not sure how to make `trend[:front] = (np.arange(0, front) * np.c_[k] + np.c_[n]).T` to work for both 1-D and 2-D, so I will fix it by squeezing if trend is 1-D.
That makes the change relatively simple and shouldn't be "dangerous" in causing another bug.

**Comment by ashukl** [comment]
Thanks, please let me know when you make the changes.

**Comment by josef-pkt** [comment]
I will finish up in a few minutes.

The unit test don't fail because the freq=2, and for freq 2 and 3 the ValueError doesn't show up. It shows up with freq=4. I'm changing the unit test to use a larger example.

**Comment by josef-pkt** [comment]
proposed fix is in #4197
I will merge it when test runs are green

**Comment by ashukl** [comment]
It seems to be working fine for me now..thanks
